cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/discovery/KubeDiscoveryService.java

```diff
@@ -54,62 +56,88 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * Implementation of {@link DiscoveryService} and {@link DiscoveryServiceClient} that uses
- * Kubernetes API to announce and discover service locations. This service assumes kubernetes
- * services are created with label, "cdap.service=[service-name]", where [service-name] is the CDAP
- * service name. On registering service via the {@link #register(Discoverable)} method, a new k8s
- * Service with name "cdap-[transformed-service-name]" will be created with label
- * "cdap.service=[service-name]". The [transformed-service-name] is the CDAP service name with "."
- * replaced with "-" to conform to the naming requirement in K8s. The service selector will be set
- * to include the current pod labels.
+ * Implementation of {@link DiscoveryService} and {@link DiscoveryServiceClient}
+ * that uses Kubernetes API to announce and discover service locations. This
+ * service assumes kubernetes services are created with label,
+ * "cdap.service=[service-name]", where [service-name] is the CDAP service name.
+ * On registering service via the {@link #register(Discoverable)} method, a new
+ * k8s Service with name "cdap-[transformed-service-name]" will be created with
+ * label "cdap.service=[service-name]". The [transformed-service-name] is the
+ * CDAP service name with "." replaced with "-" to conform to the naming
+ * requirement in K8s. The service selector will be set to include the current
+ * pod labels.
  */
-public class KubeDiscoveryService implements DiscoveryService, DiscoveryServiceClient,
+public class KubeDiscoveryService implements DiscoveryService,
+    DiscoveryServiceClient,
     AutoCloseable {
 
-  private static final Logger LOG = LoggerFactory.getLogger(KubeDiscoveryService.class);
-
-  private static final String SERVICE_LABEL = "cdap.service";
-  private static final String PAYLOAD_NAME = "cdap.service.payload";
-
+  private static final Logger LOG = LoggerFactory.getLogger(
+      KubeDiscoveryService.class);
   private static final byte[] EMPTY_PAYLOAD = new byte[0];
+  private static final String SERVICE_LABEL = "cdap.service";
+  private static final String SERVICE_TYPE_LOAD_BALANCER = "LoadBalancer";
+  private static final String SERVICE_TYPE_CLUSTER_IP = "ClusterIP";
 
   private final String namespace;
   private final String namePrefix;
-  private final Map<String, String> podLabels;
-  private final List<V1OwnerReference> ownerReferences;
   private final Map<String, DefaultServiceDiscovered> serviceDiscovereds;
   private final ApiClientFactory apiClientFactory;
   private volatile CoreV1Api coreApi;
   private volatile WatcherThread watcherThread;
   private boolean closed;
+  static final String PAYLOAD_NAME = "cdap.service.payload";
```

Comment:
use restrictive visibility unless its being used by other classes.

Reply:
Fixed and noted.